basic/src/shared/components/Button.tsx

```diff
@@ -0,0 +1,80 @@
+import { Link as RouterLink } from "react-router-dom";
+import { ClassNameValue, twJoin } from "tailwind-merge";
+import { Link, Routes } from "wasp/client/router";
+
+type ButtonSize = "md" | "sm" | "xs";
+type ButtonVariant = "primary" | "danger" | "transparent";
+
+interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  size?: ButtonSize;
+  variant?: ButtonVariant;
+}
+
+export function Button({
+  children,
+  className,
+  type = "button",
+  size = "md",
+  variant = "primary",
+  ...props
+}: ButtonProps) {
+  return (
+    <button
+      type={type}
+      className={twJoin(
+        "flex shrink-0 items-center gap-2 rounded-md font-semibold",
```

Comment:
```suggestion
        "rounded-md font-semibold",
```

I'd advise not doing layout in the `Button` component and to do the layout with a wrapper e.g. 
```tsx
<Button size="sm" onClick={deleteCompletedTasks}>
  <span className="inline-flex items-center gap-2">
    Clear completed
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 -960 960 960"
      className="size-5 fill-black"
    >
      <path d="M280-120q-33 0-56.5-23.5T200-200v-520h-40v-80h200v-40h240v40h200v80h-40v520q0 33-23.5 56.5T680-120H280Zm400-600H280v520h400v-520ZM360-280h80v-360h-80v360Zm160 0h80v-360h-80v360ZM280-720v520-520Z" />
    </svg>
  </span>
</Button>
```

This way the `Button` is concerned with who's using it.

Reply:
This once I'm a little on a fence.
If this was a generic/abstract button, sure. 

But here we have styled button component meant to be used in a specific way. 
Ensuring the content behaves with what's expected of this button is okay I think.

Reply:
> But here we have styled button component meant to be used in a specific way.

Let's maybe then encode that specific expectation explicitly. Right now you have `flex` styles that apply to both scenarios, with and without icons. And it's okay now and nothing breaks... but it's a button that users can reuse afterwards in their app. And this implicit `flex` styling might cause some trouble for them.

Two options:
1. I'd either remove the styling and do the layout as I suggested in my initial comment
2. Or, we could go for an explicit `icon` or `iconLeft` / `iconRight` prop that would render the icon + make sure the style matches for that scenario.

Reply:
Hmm I did:
3)
```ts
 <Button
    className="flex items-center gap-2"
    // ...
``` 
Where it is needed. I really dislike the `icon` approach. But if we don't want to force flex then maybe use what we already provide.